test-result-summary-client/src/TestCompare/TestCompare.jsx

```diff
@@ -33,29 +35,105 @@ export default class TestCompare extends Component {
         this.setState( this.state );
     };
 
+    toggleTimeStamp = () => {
+        this.setState({ timeStamp: !this.state.timeStamp });
+    }
+
+    toggleDeepSmith = () => {
+        this.setState({ deepSmith: !this.state.deepSmith });
+    }
+
     handleCompare = async () => {
         $( this.compare ).mergely( 'lhs', '' );
         $( this.compare ).mergely( 'rhs', '' );
 
-        for ( var i = 0; i < this.state.forms.length; i++ ) {
+        for ( var i = 0; i < this.state.forms.length; i++ ) {
             const { url, buildName, buildNum, testName } = this.state.forms[i];
             const response = await fetch( `/api/getOutputByTestInfo?url=${encodeURIComponent( url )}&buildName=${encodeURIComponent( buildName )}&buildNum=${buildNum}&testName=${testName}`, {
                 method: 'get'
             } );
             const res = await response.json();
             if ( res && res.output ) {
+                let curOutput = res.output;
+                if (this.state.timeStamp === false) {
+                    curOutput = this.removeTimeStamp(curOutput);
+                }
+                if (this.state.deepSmith === true) {
+                    curOutput = this.matchDeepSmith(curOutput);
+                }     
+                res.output = curOutput;
                 this.state.tests[i] = res;
             } else {
                 alert( "Cannot find data! Please check your input values." );
                 return;
-            }
+            }
         }
         this.setState( this.state );
 
         $( this.compare ).mergely( 'lhs', this.state.tests[0].output );
         $( this.compare ).mergely( 'rhs', this.state.tests[1].output );
     }
 
+    removeTimeStamp = (inputText) => {
+        return inputText.replace(/\[\d{4}-\d{2}-\d{2}.*?\] /g, "");
+    }
+
+    matchDeepSmith = (inputText) => {
+        // remove beginning and bottom build info
+        let startWords = "Running Java Driver:";

```

Comment:
`const` should be used for a variable that does not change. Same for `endWords `, `testStartWords `